src/posix.h

```diff
@@ -0,0 +1,57 @@
+#ifndef BENCHMARK_POSIX_H_
+#define BENCHMARK_POSIX_H_
+
+#include <stdlib.h>
+#include <stdio.h> // fileno, _fileno
+#include <string.h>  // _stricmp
+
+#include "internal_macros.h"
+
+#ifdef OS_WINDOWS
+#include <io.h>  // _isatty
+#else
+#include <strings.h>  // strcasecmp
+#include <unistd.h>   // isatty
+#endif
+
+namespace benchmark {
+namespace posix {
```

Comment:
I know the top level namespace `posix` is reserved for the implementation in C++. I would prefer not using a nested one either. Could you come up with another name?